src/App.tsx

```diff
@@ -1,14 +1,78 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [statusFilter, setStatusFilter] = useState('all');
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const loadTodos = async () => {
+    setIsLoading(true);
+    const todosFromApi = await getTodos();
+    setTodos(todosFromApi);
+    setFilteredTodos(todosFromApi);
+    setIsLoading(false);
+  };
+
+  const handleStatusChange = (status: string) => {
+    setStatusFilter(status);
+    filterTodos(status, searchQuery);
+  };
+
+  const handleSearchChange = (query: string) => {
+    setSearchQuery(query);
+    filterTodos(statusFilter, query);
+  };
+
+  const filterTodos = (status: string, query: string) => {
+    let updatedTodos = todos;
+
+    if (status !== 'all') {
+      updatedTodos = todos.filter(todo =>
+        status === 'completed' ? todo.completed : !todo.completed
+      );
+    }
+
+    if (query) {
+      updatedTodos = updatedTodos.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase())
+      );
+    }
+
+    setFilteredTodos(updatedTodos);
+  };
+
+  const openTodoModal = async (todo: Todo) => {
+    setIsLoading(true);
+    const user = await getUser(todo.userId);
```

Comment:
Consider adding error handling for the `getUser` API call. This will ensure that any errors during the fetch operation are managed properly, preventing the application from crashing and allowing you to provide feedback to the user.